packages/frontend-2/components/viewer/explorer/StringFilterItem.vue

```diff
@@ -122,6 +128,18 @@ const color = computed(() => {
     ?.color
 })
 
+const highlightObject = () => {
```

Comment:
A bit too low level for a Vue component IMO, these should be moved to a `useHighlightedObjectsUtilities` composable in `lib/viewer/composables/ui.ts`, so that you can easily highlight/un-highlight objects without having to do low level array manipulation. Not only that, but you're duplicating these same array manipulations in each Vue component.

Additionally, there could be a shit ton of IDs in these arrays, so we have to be careful with how we do the highlighting/un-highlighting and we have to ensure its performant. So test all of this highlighting out w/ large models where you're potentially highlighting large amounts of IDs at the same time.

Reply:
I tested the highlighting with a large Rhino file on my computer. There's a slight slowdown, but the highlighting itself doesn’t add much to the overall lag. The viewer is generally less responsive due to the model size, but the highlighting function works well even with many IDs.